modules/programs/fish.nix

```diff
@@ -340,41 +340,25 @@ in {
         # ~/.config/fish/config.fish: DO NOT EDIT -- this file has been generated
         # automatically by home-manager.
 
-        # if we haven't sourced the general config, do it
-        if not set -q __fish_general_config_sourced
-
-          set --prepend fish_function_path ${
-            if pkgs ? fishPlugins && pkgs.fishPlugins ? foreign-env then
-              "${pkgs.fishPlugins.foreign-env}/share/fish/vendor_functions.d"
-            else
-              "${pkgs.fish-foreign-env}/share/fish-foreign-env/functions"
-          }
-          fenv source ${config.home.profileDirectory}/etc/profile.d/hm-session-vars.sh > /dev/null
-          set -e fish_function_path[1]
-
-          ${cfg.shellInit}
-          # and leave a note so we don't source this config section again from
-          # this very shell (children will source the general config anew)
-          set -g __fish_general_config_sourced 1
+        set --prepend fish_function_path ${
+          if pkgs ? fishPlugins && pkgs.fishPlugins ? foreign-env then
+            "${pkgs.fishPlugins.foreign-env}/share/fish/vendor_functions.d"
+          else
+            "${pkgs.fish-foreign-env}/share/fish-foreign-env/functions"
```

Comment:
While at it, you can also remove these lines and the `if`.
We don't need the compatibility anymore at this point.

Reply:
I was going to get my hands on these too, thanks for the pointer! However I think we'd better do it in a separate PR since I'd like to add some tips for Fish users to README as well.

Reply:
My mistake, I meant to say that you should remove the if logic and preserve the `pkgs.fishPlugins.foreign-env`.

Reply:
Got it. But I guess the tips are harmless so I will keep it.